### DISCUSSION TITLE
Eclipse JDT need add-opens and add-modules?

### DISCUSSION THREAD

**Discussion by jreicher** [joaotavora/eglot#911] [Q&A] [answered]
Using jdt-1.9.0 and eglot both out of the box on Windows, the language server can't resolve names such as Object and String.

The jdt docs say to run it with

...
--add-modules=ALL-SYSTEM
--add-opens java.base/java.util=ALL-UNNAMED
--add-opens java.base/java.lang=ALL-UNNAMED
...

and this does indeed fix the issue. Is there a reason this isn't part of the arguments hardcoded for eglot-eclipse-jdt? Is there a different/better way to have jdt resolve these names?

**Comment by joaotavora** [comment, marked as answer]
What is `eglot-eclipse-jdt` for you?  That former class has been removed from stock Eglot, though you can add it in your .emacs if you see the need.

In a recent Eglot, `eglot-eclipse-jdt` has been replaced by a argumentless call to a new Python startup script in JDT 1.9, `jdtls`. See https://github.com/joaotavora/eglot/discussions/868.  So maybe:

* it's that Python script that should have sensible defaults built in.  
* maybe Eglot's default invocation of that startup script should have some arguments, but that Java freight train is not something I'd like to see in eglot.el.  So I'm more inclined to putting those in JDT python startup script.

**Reply by jreicher** [reply]
Ah, thanks; I was using eglot 1.8, from elpa. I see the dev version uses that python script, and the script does indeed have those arguments in there.

**Reply by joaotavora** [reply]
Great.  I've marked this as the answer.
